File: problema 16/fortran/problema_16.f90
program PythagoreanTriples
    implicit none
    integer :: a, b, c
    integer :: c2
    integer, dimension(3) :: sides

    do a = 1, 500
        do b = a, 500
            c2 = a**2 + b**2
            c = sqrt(real(c2))
            if (c**2 == c2 .and. c <= 500) then
                sides(1) = a
                sides(2) = b
                sides(3) = c
                print *, "las ternas son: ", sides(1), sides(2), sides(3)
            end if
        end do
    end do
end program PythagoreanTriples

File: problema 16/problema_16.c
#include <stdio.h>
#include <math.h>

int main() {
    int a, b, c, c2;
    int lados[3]; // Arreglo para almacenar los lados del triángulo

    //calcular los lados de las ternas
        for ( a = 1; a <= 500; a++){
            for ( b = a; b <= 500; b++){
                c2 = a * a + b * b;
                c = sqrt(c2);
                if (c *c == c2 && c<= 500){
                    lados[0] = a;
                    lados[1] = b;
                    lados[2] = c;
                    printf("las ternas son: %d, %d, %d\n", lados[0], lados[1], lados[2]);
            }
        }
        return 0;
    }
}
